openstef_dbc/services/predictions.py

```diff
@@ -71,6 +71,132 @@ def get_predicted_load(
+            |> range(start: {bind_params["_start"].strftime('%Y-%m-%dT%H:%M:%SZ')}, stop: {bind_params["_stop"].strftime('%Y-%m-%dT%H:%M:%SZ')}) 
+            |> filter(fn: (r) => 
+                r._measurement == "prediction")
+            |> filter(fn: (r) => 
+                r._field == "quality" )                 
+            |> filter(fn: (r) => r.pid == "{bind_params["pid"]}")
+        """
+
+        # Query the database
+        result = _DataInterface.get_instance().exec_influx_query(query, bind_params)
+
+        # For multiple Fields a list is returned.
+        if isinstance(result, list):
+            result = pd.concat(result)[["_value", "_field", "_time"]]
+
+        # Return result
+        if not result.empty:
+            # Shifting is needed to output the same as with the old influx client
+            return parse_influx_result(result, aggfunc="first").shift(-15, freq="T")
+        else:
+            return pd.Series()
+
+    def get_prediction_including_components(
+        self,
+        pj: dict,
+        start_time: datetime = None,
+        end_time: datetime = None,
+        quantiles: bool = False,
+    ):
+        """Get historic load predictions for given pid including component forecasts.
+
+        Args:
+            pj (dict): Prediction job
+            start_time (datetime): Start time  to retrieve the historic load prediction.
+            end_time (datetime): End timeto retrieve the historic load prediction.
+            quantiles: (boo): Indicates wheter quantiles should be retrieved as well.
+
+        Returns:
+            pandas.DataFrame: Dataframe with the total forecast,
+            the components forecasts and the quantiles if they are requested.
+
+        """
+        # Apply default parameters if none are provided
+        if start_time is None:
+            start_time = datetime.utcnow()
+        if end_time is None:
+            end_time = datetime.utcnow() + timedelta(days=2)
+
+        bind_params = {
+            "pid": str(pj["id"]),
+            "_start": start_time.astimezone(pytz.UTC),
+            "_stop": end_time.astimezone(pytz.UTC),
+        }
+        quantile_section = (
+            """or r._field == "quantile_P"""
+            + """" or r._field == "quantile_P""".join(
+                [f"{quantile * 100:02.0f}" for quantile in pj["quantiles"]]
+            )
+            + '"'
+        )
+
+        if quantiles:
+            query = f"""
+            from(bucket: "forecast_latest/autogen")
+                |> range(start: {bind_params["_start"].strftime('%Y-%m-%dT%H:%M:%SZ')}, stop: {bind_params["_stop"].strftime('%Y-%m-%dT%H:%M:%SZ')}) 
+                |> filter(fn: (r) => 
+                    r._measurement == "prediction")
+                |> filter(fn: (r) => 
+                   r._field == "forecast" or r._field == "stdev" or r._field == "forecast_other" or r._field == "forecast_solar" or r._field == "forecast_wind_on_shore"{quantile_section})                 
```

Comment:
Perhaps specifying these fields as a list somewhere makes it more easy to spot and potentially change them in future.